pkg/cmd/roachtest/roachtestutil/mixedversion/planner.go

```diff
@@ -1362,6 +1363,37 @@ func (ss stepSelector) Filter(predicate func(*singleStep) bool) stepSelector {
 	return result
 }
 
+// Cut returns the before and after of a cut step selector, as well as the step that was cut out.
```

Comment:
nit: what is a cut step selector? What does returning the before and after mean? What if no steps match the predicate? Lets be more specific:

```
// `Cut` finds the first step that matches the predicate given, returning
// new selectors for the steps before and after the matching step. It also
// returns a new selector for the matching cut step. If no step matches the
// predicate, `Cut` returns `ss`, nil, nil.